### PR TITLE
Gatsby Storybook POC: test basic args and jsdoc on a handful of components

### PR DESCRIPTION
**Pull request by danidewitt**
This PR is a ***POC*** meant to test v6.5 Storybook arguments in Gatsby with [story.to.design](https://story.to.design/) and documenting with [jsdoc](https://jsdoc.app/). Only a select number of components were selected on purpose. 

We can automate this (as shown below), but this PR of adding basic jsdocs and args is meant to be a faster way to POC before dialing in the automation further. 

```
import React from "react";
import { storiesOf } from "@storybook/react";
import * as ezComponents from "../src/components";
import { extractArgTypes } from "@storybook/react/dist/cjs/client/docs/extractArgTypes";

const defaultArgs = {
  EzButton: { children: <span>click me!</span> },
  EzWell: { children: <div>hi, well!</div> },
};

const names = [];
for (const componentName in ezComponents) {
  names.push(componentName);
}
names.sort();

for (const componentName of names) {
  if (componentName.startsWith("Ez")) {
    const Component = ezComponents[componentName];
    const stories = storiesOf(`Figma/${componentName}`, module);
    const metadataFromArgTypes = extractArgTypes(Component);
    console.log({ metadataFromArgTypes });
    stories.add(componentName, (args: any) => <Component {...args} />, {
      component: Component,
      args: defaultArgs[componentName] || {},
      argTypes: metadataFromArgTypes,
      parameters: {
        theme: "fulfillment",
      },
    });
  }
}
```
 
We could also automate the rest of the docs we have in `.md` files by extending `all.stories.tsx`.

## What did we change?
- Added basic arguments to a select number of components
- Added ezCater icons for Storybook usage

## Why are we doing this?
To get a basic version of Storybook working with arguments so we can decide if we want to buy story.to.design.

## Screenshot(s) / Gif(s):
Screenshot 2023-08-17 at 10 01 49 AM
Screenshot 2023-08-17 at 10 02 00 AM
Screenshot 2023-08-17 at 10 01 24 AM

## Checklist

- [ ] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes
 
[Contributing guidelines](https://ezcater.github.io/recipe/guides/contributing/)

### CODE REVIEW

File: packages/recipe/package.json [hunk 1]
```diff
@@ -94,12 +95,12 @@
     "react-dom": "^17.0.0 || ^18.0.0"
   },
   "scripts": {
-    "build-storybook": "build-storybook",
+    "build-storybook": "rimraf \"storybook-static/**/*\" ../../docs/static/storybook && build-storybook",
```
Comment (danidewitt) [inline packages/recipe/package.json:98]: i'm aware this command isn't working in CI. i'm working on it.

**Comment by noranda** [comment]
Were the args you added in this PR automated or were they manually added? What was the POC?

We've already manually added all arg types up to EzField in the `storybook-docs` branch. We could pull over those arg types to the `6.5` storybook here, but I'm not sure what value this gets us. It might be more valuable to work on adding the remaining arg types to that branch (after it's merged into `main`), especially since we should hopefully have a public link by today or tomorrow.

My concern is that we don't want this to hold up our `v18` release. Once we have the public link, we'll be creating a new `v17` or `gatsby` branch, before making a PR for the release. If you'd like to explore this further after the release, that works for me.

**Comment by noranda** [comment]
This PR will need to change the branch it merges into to be `gatsby-docs`.

**Review by noranda — APPROVED** [review]
I am interested to see if the release changes we made for the `gatsby-docs` site work as expected. I don't think changesets are required for this branch anymore, but they shouldn't hurt anything.

**Review by pgrippi — APPROVED** [review]

**Review by pgrippi — APPROVED** [review]

**Pull request merged**